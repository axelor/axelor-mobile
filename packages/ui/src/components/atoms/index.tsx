/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

export {default as BlockInteractionScreen} from './BlockInteractionScreen/BlockInteractionScreen';
export {default as Card} from './Card/Card';
export {default as HorizontalRule} from './HorizontalRule/HorizontalRule';
export {default as HtmlInput} from './HtmlInput/HtmlInput';
export {default as DottedLine} from './DottedLine/DottedLine';
export {default as Icon} from './Icon/Icon';
export {default as Input} from './Input/Input';
export {default as Screen} from './Screen/Screen';
export {default as ScrollView} from './ScrollView/ScrollView';
export {default as Switch} from './Switch/Switch';
export {default as Text} from './Text/Text';
