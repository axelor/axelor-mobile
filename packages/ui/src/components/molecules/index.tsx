/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

export {default as AttachmentCard} from './AttachmentCard/AttachmentCard';
export {default as Badge} from './Badge/Badge';
export {default as BootstrapIcon} from './BootstrapIcon/BootstrapIcon';
export {default as Button} from './Button/Button';
export {default as CardIconButton} from './CardIconButton/CardIconButton';
export {default as Checkbox} from './Checkbox/Checkbox';
export {default as Chip} from './Chip/Chip';
export {default as CircleButton} from './CircleButton/CircleButton';
export {default as ClearableCard} from './ClearableCard/ClearableCard';
export {default as DropdownCard} from './DropdownCard/DropdownCard';
export {default as DropdownMenu} from './DropdownMenu/DropdownMenu';
export {default as DropdownMenuItem} from './DropdownMenuItem/DropdownMenuItem';
export {default as EditableInput} from './EditableInput/EditableInput';
export {default as FormHtmlInput} from './FormHtmlInput/FormHtmlInput';
export {default as FormIncrementInput} from './FormIncrementInput/FormIncrementInput';
export {default as FormInput} from './FormInput/FormInput';
export {default as FromTo} from './FromTo/FromTo';
export {default as HalfLabelCard} from './HalfLabelCard/HalfLabelCard';
export {default as IconButton} from './IconButton/IconButton';
export {default as IconInput} from './IconInput/IconInput';
export {default as Image} from './Image/Image';
export {default as Increment} from './Increment/Increment';
export {default as InfoBubble} from './InfoBubble/InfoBubble';
export {default as HeaderIndicator} from './HeaderIndicator/HeaderIndicator';
export {default as LabelText} from './LabelText/LabelText';
export {default as LoadingIndicator} from './LoadingIndicator/LoadingIndicator';
export {default as MessageBox} from './MessageBox/MessageBox';
export {default as MovementIndicationCard} from './MovementIndicationCard/MovementIndicationCard';
export {default as MultiValuePickerButton} from './MultiValuePickerButton/MultiValuePickerButton';
export {default as NotesCard} from './NotesCard/NotesCard';
export {default as PanelTabs} from './PanelTabs/PanelTabs';
export {default as PopUp} from './PopUp/PopUp';
export {default as RightIconButton} from './RightIconButton/RightIconButton';
export {default as SelectionContainer} from './SelectionContainer/SelectionContainer';
export {default as StarScore} from './StarScore/StarScore';
export {default as SwitchCard} from './SwitchCard/SwitchCard';
export {default as TitledValue} from './TitledValue/TitledValue';
export {default as ToggleSwitch} from './ToggleSwitch/ToggleSwitch';
export {default as UnorderedList} from './UnorderedList/UnorderedList';
export {default as ViewAllContainer} from './ViewAllContainer/ViewAllContainer';
