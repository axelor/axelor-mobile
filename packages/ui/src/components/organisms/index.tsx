/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2024 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

export {default as ActionCard} from './ActionCard/ActionCard';
export {default as AutoCompleteSearch} from './AutoCompleteSearch/AutoCompleteSearch';
export {default as CheckboxScrollList} from './CheckboxScrollList/CheckboxScrollList';
export {default as ChipSelect} from './ChipSelect/ChipSelect';
export {default as DistributionBar} from './DistributionBar/DistributionBar';
export {default as DraggableWrapper} from './DraggableWrapper/DraggableWrapper';
export {default as DropdownCardSwitch} from './DropdownCardSwitch/DropdownCardSwitch';
export {default as DropdownMenuItem} from './DropdownMenuItem/DropdownMenuItem';
export {default as DurationInput} from './DurationInput/DurationInput';
export {default as FloatingButton} from './FloatingButton/FloatingButton';
export {default as GridView} from './GridView/GridView';
export {default as GroupByScrollList} from './GroupByScrollList/GroupByScrollList';
export {default as HeaderContainer} from './HeaderContainer/HeaderContainer';
export {default as ImageBubble} from './ImageBubble/ImageBubble';
export {default as InfoBubble} from './InfoBubble/InfoBubble';
export {default as InfoButton} from './InfoButton/InfoButton';
export {default as MultiValuePicker} from './MultiValuePicker/MultiValuePicker';
export {default as Picker} from './Picker/Picker';
export {default as ProgressBar} from './ProgressBar/ProgressBar';
export {default as QuantityCard} from './QuantityCard/QuantityCard';
export {ActionType, ScrollList, TopActions} from './ScrollList';
export {default as SearchBar} from './SearchBar/SearchBar';
export {default as TagList} from './TagList/TagList';
export {default as ToggleButton} from './ToggleButton/ToggleButton';
export {default as ViewAllEditList} from './ViewAllEditList/ViewAllEditList';
