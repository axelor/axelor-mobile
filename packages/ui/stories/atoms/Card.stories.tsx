/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import {View, StyleSheet} from 'react-native';
import {storiesOf} from '@storybook/react-native';
import {Card, Text} from '../../src/components/atoms';

storiesOf('ui/atoms/Card', module)
  .addDecorator(story => <View style={styles.decorator}>{story()}</View>)
  .add('Default', () => (
    <Card>
      <View />
    </Card>
  ))
  .add('Text', () => (
    <Card>
      <Text>Text</Text>
    </Card>
  ))
  .add(
    'Custom Style',
    args => (
      <Card {...args}>
        <View style={styles.defaultCustom} />
      </Card>
    ),
    {
      argTypes: {
        style: {
          control: {
            type: 'object',
          },
          defaultValue: {
            width: 0,
            height: 5,
            backgroundColor: 'green',
          },
        },
      },
    },
  );

const styles = StyleSheet.create({
  decorator: {
    padding: 20,
  },
  defaultCustom: {
    height: 100,
    backgroundColor: '#B4503B',
  },
});
