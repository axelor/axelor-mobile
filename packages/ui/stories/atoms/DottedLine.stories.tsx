/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import {View} from 'react-native';
import {storiesOf} from '@storybook/react-native';
import {DottedLine} from '../../src/components/atoms';

storiesOf('ui/atoms/DottedLine', module).add(
  'default',
  args => (
    <View {...args}>
      <DottedLine />
    </View>
  ),
  {
    argTypes: {
      style: {
        control: {
          type: 'object',
        },
        defaultValue: {
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
        },
      },
    },
  },
);
