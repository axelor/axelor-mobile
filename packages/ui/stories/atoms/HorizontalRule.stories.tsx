/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import {StyleSheet} from 'react-native';
import {storiesOf} from '@storybook/react-native';
import {HorizontalRule, Text} from '../../src/components/atoms';

storiesOf('ui/atoms/HorizontalRule', module).add('default', () => (
  <>
    <Text>Before rule</Text>
    <HorizontalRule style={styles.container} />
    <Text>After rule</Text>
  </>
));

const styles = StyleSheet.create({
  container: {
    marginVertical: 10,
  },
});
