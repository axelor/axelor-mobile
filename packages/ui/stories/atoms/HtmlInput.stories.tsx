/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import {View, StyleSheet} from 'react-native';
import {storiesOf} from '@storybook/react-native';
import {HtmlInput} from '../../src/components/atoms';

storiesOf('ui/atoms/HtmlInput', module).add(
  'Default',
  args => {
    return (
      <View style={styles.container}>
        <HtmlInput
          title="Description"
          defaultInput="<p>Hello World!<b>This text is bold.</b></p>"
          onChange={console.log}
          {...args}
        />
      </View>
    );
  },
  {
    argTypes: {
      style: {control: 'object'},
      styleToolbar: {control: 'object'},
      containerStyle: {control: 'object'},
      editorBackgroundColor: {control: 'color'},
      title: {control: 'text'},
      placeholder: {control: 'text'},
      defaultInput: {control: 'text'},
      readonly: {control: 'boolean'},
      onHeightChange: {action: 'onHeightChange'},
    },
  },
);

const styles = StyleSheet.create({
  container: {
    marginTop: 50,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
