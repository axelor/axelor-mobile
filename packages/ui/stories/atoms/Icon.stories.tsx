/*
 * Axelor Business Solutions
 *
 * Copyright (C) 2023 Axelor (<http://axelor.com>).
 *
 * This program is free software: you can redistribute it and/or  modify
 * it under the terms of the GNU Affero General Public License, version 3,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {Icon} from '../../src/components/atoms';
import {lightTheme} from '../../src/theme';

const stories = storiesOf('ui/atoms/Icon', module);

stories.add(
  'Default',
  args => {
    return (
      <Icon
        {...args}
        color={lightTheme.colors[args.color].background}
        FontAwesome5={false}
      />
    );
  },
  {
    argTypes: {
      size: {
        control: {type: 'number', min: 10, max: 50},
        defaultValue: 20,
      },
      name: {
        type: 'string',
        defaultValue: 'heart',
        control: {type: 'text'},
      },
      touchable: {
        type: 'boolean',
      },
      visible: {
        type: 'boolean',
      },
      color: {
        options: Object.entries(lightTheme.colors)
          .filter(([, _color]) => typeof _color !== 'string')
          .map(([key]) => key),
        defaultValue: 'plannedColor',
        control: {
          type: 'select',
        },
      },
    },
  },
);
