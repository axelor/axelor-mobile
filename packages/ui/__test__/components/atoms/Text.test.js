import React from 'react';
import {shallow} from 'enzyme';
import {Text} from '../../../lib/components/atoms';
import {Text as ReactNativeText} from 'react-native';

describe('Text Component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Text />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const text = 'Hello, World!';
    const wrapper = shallow(<Text>{text}</Text>);

    expect(wrapper.find(ReactNativeText).length).toBe(1);
    expect(wrapper.find(ReactNativeText).children().text()).toBe(text);
  });

  it('applies custom styles correctly', () => {
    const customStyle = {color: 'red', fontSize: 20};
    const wrapper = shallow(<Text style={customStyle}>Custom Text</Text>);

    expect(wrapper.prop('style')).toContain(customStyle);
  });

  it('applies the specified number of lines', () => {
    const numberOfLines = 2;
    const wrapper = shallow(
      <Text numberOfLines={numberOfLines}>Long Text</Text>,
    );

    expect(wrapper.prop('numberOfLines')).toBe(numberOfLines);
  });

  it('adjusts font size to fit when enabled', () => {
    const wrapper = shallow(<Text adjustsFontSizeToFit>Resizable Text</Text>);

    expect(wrapper.prop('adjustsFontSizeToFit')).toBe(true);
  });

  it('invokes onTextLayout callback', () => {
    const onTextLayout = jest.fn();
    const wrapper = shallow(
      <Text onTextLayout={onTextLayout}>Text with Layout</Text>,
    );

    wrapper.simulate('textLayout', {nativeEvent: {layout: {height: 100}}});

    expect(onTextLayout).toHaveBeenCalledTimes(1);
    expect(onTextLayout).toHaveBeenCalledWith({
      nativeEvent: {layout: {height: 100}},
    });
  });
});
