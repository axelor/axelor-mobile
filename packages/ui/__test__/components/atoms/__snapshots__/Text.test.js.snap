// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders correctly 1`] = `ShallowWrapper {}`;
